deepmd/pt/model/descriptor/dpa3.py

```
@@ -469,16 +469,18 @@ def forward(
         # cast the input to internal precsion
         extended_coord = extended_coord.to(dtype=self.prec)
         nframes, nloc, nnei = nlist.shape
-        nall = extended_coord.view(nframes, -1).shape[1] // 3
-
-        node_ebd_ext = self.type_embedding(extended_atype)
-        node_ebd_inp = node_ebd_ext[:, :nloc, :]
+        # nall = extended_coord.view(nframes, -1).shape[1] // 3
+        if comm_dict is None:
+            atype = extended_atype[:, :nloc]
+        else:
+            atype = extended_atype
```

Ruff (0.8.2)

Use ternary operator `atype = extended_atype[:, :nloc] if comm_dict is None else extended_atype` instead of `if`-`else`-block

Replace `if`-`else`-block with `atype = extended_atype[:, :nloc] if comm_dict is None else extended_atype`

(SIM108)